src/index.js

```diff
@@ -26,7 +26,8 @@ export default function mitt(all: EventHandlerMap) {
 		 * @memberOf mitt
 		 */
 		on(type: string, handler: EventHandler) {
-			(all[type] || (all[type] = [])).push(handler);
+			var arr = all[type] = all[type] || [];
+			return arr.splice.bind(arr, arr.push(handler) - 1, 1);
```

Comment:
Good point, my "late night coding" brain missed how brittle the splice solution was.